tests/table/test_expire_snapshots.py

```diff
@@ -223,3 +227,384 @@ def test_expire_snapshots_by_ids(table_v2: Table) -> None:
     assert EXPIRE_SNAPSHOT_1 not in remaining_snapshots
     assert EXPIRE_SNAPSHOT_2 not in remaining_snapshots
     assert len(table_v2.metadata.snapshots) == 1
+
+
+def test_thread_safety_fix() -> None:
+    """Test that ExpireSnapshots instances have isolated state."""
+    # Create two mock transactions (representing different tables)
+    transaction1 = Mock()
+    transaction2 = Mock()
+
+    # Create two ExpireSnapshots instances
+    expire1 = ExpireSnapshots(transaction1)
+    expire2 = ExpireSnapshots(transaction2)
```

Comment:
To simplify the tests, we can also just pass in a `None` transaction, since we never call the `.commit()` method on the operation.
```suggestion
    # Create two ExpireSnapshots instances
    expire1 = ExpireSnapshots(None)
    expire2 = ExpireSnapshots(None)
```